homeassistant/components/tractive/switch.py

```diff
@@ -0,0 +1,167 @@
+"""Support for Tractive switches."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+import logging
+from typing import Any
+
+from aiotractive.exceptions import TractiveError
+
+from homeassistant.components.switch import SwitchEntity, SwitchEntityDescription
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .const import (
+    ATTR_BUZZER,
+    ATTR_LED,
+    ATTR_LIVE_TRACKING,
+    CLIENT,
+    DOMAIN,
+    SERVER_UNAVAILABLE,
+    TRACKABLES,
+    TRACKER_HARDWARE_STATUS_UPDATED,
+)
+from .entity import TractiveEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class TractiveSwitchEntityDescription(SwitchEntityDescription):
+    """Class describing Tractive switch entities."""
+
+    method: str | None = None
+
+
+SWITCH_TYPES = (
+    TractiveSwitchEntityDescription(
+        key=ATTR_BUZZER,
+        name="Tracker Buzzer",
+        icon="mdi:volume-high",
+        method="async_set_buzzer",
+    ),
+    TractiveSwitchEntityDescription(
+        key=ATTR_LED,
+        name="Tracker LED",
+        icon="mdi:led-on",
+        method="async_set_led",
+    ),
+    TractiveSwitchEntityDescription(
+        key=ATTR_LIVE_TRACKING,
+        name="Live Tracking",
+        icon="mdi:map-marker-path",
+        method="async_set_live_tracking",
+    ),
+)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up Tractive switches."""
+    client = hass.data[DOMAIN][entry.entry_id][CLIENT]
+    trackables = hass.data[DOMAIN][entry.entry_id][TRACKABLES]
+
+    entities = []
+
+    for item in trackables:
+        for description in SWITCH_TYPES:
+            entities.append(
+                TractiveSwitch(
+                    client.user_id,
+                    item.trackable,
+                    item.tracker,
+                    item.tracker_details,
+                    f"{item.trackable['_id']}_{description.key}",
```

Comment:
Why construct the `unique_id` here? You already need to assign `self._attr_unique_id` and both, `item.trackable` and `description`, are passed to `__init__`.